### PR TITLE
[v2, windows] Add WebviewUserDataPath to windows options

### LINKED ISSUE #1103: [v2, windows] Random panics during processRequest
**Description**
If there's a lot of WebResource processing ongoing some random panics occur. It seems like some COM syscalls return a nil pointer and don't set the pointer to the pointer onto the Go stack.

```go
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x607a94]

goroutine 1 [running, locked to thread]:
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceRequest).Release(0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequest.go:45 +0x14
panic({0x895200, 0xd37b70})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/runtime/panic.go:1038 +0x215
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceRequest).GetUri(0x7ff8b67a8970)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequest.go:32 +0x3c
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).processRequest(0xc00012e3f0, 0xc00039d3e8, 0x7b7a000a1560)
	/builds/app/vendor/github.com/wailsapp/wails/v2/internal/frontend/desktop/windows/frontend.go:349 +0x55
github.com/leaanthony/go-webview2/pkg/edge.(*Chromium).WebResourceRequested(0xc000388000, 0xc00039d530, 0xc00039d4a8)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/chromium.go:260 +0xbd
github.com/leaanthony/go-webview2/pkg/edge._ICoreWebView2WebResourceRequestedEventHandlerInvoke(0xc00039d4b8, 0x462c9d, 0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequestedEventHandler.go:26 +0x22
syscall.Syscall(0x7ff911e31a60, 0x1, 0x15c1810, 0x0, 0x0)
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/runtime/syscall_windows.go:483 +0xf4
syscall.(*Proc).Call(0x40d767, {0xc0003754b0, 0x873de0, 0x2bdd9e01})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/syscall/dll_windows.go:183 +0xc6
syscall.(*LazyProc).Call(0xc000155650, {0xc0003754b0, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/syscall/dll_windows.go:340 +0x50
github.com/leaanthony/winc/w32.DispatchMessage(...)
	/builds/app/vendor/github.com/leaanthony/winc/w32/user32.go:327
github.com/leaanthony/winc.RunMainLoop()
...
```

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x6041b4]

goroutine 1 [running, locked to thread]:
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceResponse).Release(0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceResponse.go:25 +0x14
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).processRequest(0xc0001682d0, 0xc000107390, 0x1b06000b0780)
	/builds/app/vendor/github.com/wailsapp/wails/v2/internal/frontend/desktop/windows/frontend.go:400+0x519
github.com/leaanthony/go-webview2/pkg/edge.(*Chromium).WebResourceRequested(0xc00006af00, 0xc0001074d8, 0xc000107450)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/chromium.go:260 +0xbd
github.com/leaanthony/go-webview2/pkg/edge._ICoreWebView2WebResourceRequestedEventHandlerInvoke(0xc000107460, 0x46299d, 0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequestedEventHandler.go:26 +0x22
syscall.Syscall(0x7ff954bc1a60, 0x1, 0x12924e0, 0x0, 0x0)
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/runtime/syscall_windows.go:483 +0xf4
syscall.(*Proc).Call(0x40d507, {0xc0003907c0, 0x86b1a0, 0x2be64601})
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/syscall/dll_windows.go:183 +0xc6
syscall.(*LazyProc).Call(0xc00017f680, {0xc0003907c0, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/syscall/dll_windows.go:340 +0x50
github.com/leaanthony/winc/w32.DispatchMessage(...)
	/builds/app/vendor/github.com/leaanthony/winc/w32/user32.go:327
github.com/leaanthony/winc.RunMainLoop()
....
```

**To Reproduce**
Steps to reproduce the behaviour: Create a demo app with several resources and continuously reload the window, e.g. every second.

**Additional context**
After some investigation I've found out that `go-webview2` doesn't satisfy all Go [unsafe.Pointer](https://pkg.go.dev/unsafe#Pointer) rules. I've provided PR leaanthony/go-webview2#7 to fix those issues.

### LINKED ISSUE #1104: [v2, windows] Option to specify the userdata folder of WebView2
**Is your feature request related to a problem? Please describe.**
At the moment the path where the webview2 stores its user data is hardcoded, `%APPDATA%\[BinaryName.exe]`. If an application already stores some user data in another folder, there's no possibility to let the webview store its data in the same folder.

**Describe the solution you'd like**
Having a field `WebviewUserDataPath` in the windows options to specify the folder to be used by the webview for storing its user data.

**Describe alternatives you've considered**
There's no workaround other than using the folder `%APPDATA%\[BinaryName.exe]` to store the custom user data. Sometimes this folder isn't the preferred choice of the developer.

**Additional context**
The functionality to pass the DataPath has already been implemented in go-webview2 upstream. I've created PR leaanthony/go-webview2#6 to merge the changes from upstream.

### PR DESCRIPTION
**Pull request by stffabi**
Fix #1104
Fix #1103

### CODE REVIEW

File: v2/go.mod [hunk 1]
```diff
@@ -80,3 +80,5 @@ require (
 	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 )
+
+replace github.com/leaanthony/go-webview2 v1.0.2 => github.com/stffabi/go-webview2 v0.0.0-20220121103731-36d379a830f1
```
Comment (stffabi) [inline v2/go.mod:84]: Will be replaced as soon as leaanthony/go-webview2#6 has been merged.

**Comment by stffabi** [comment]
@leaanthony  I've bumped go-webview2 to master and removed the replace. So we are good to go.

**Comment by leaanthony** [comment]
Just have one question in the review

**Comment by leaanthony** [comment]
Just wondering what would happen if the data path was invalid or blank. Should we do checks?

**Comment by stffabi** [comment]
If it's blank then the default will be applied in go-webview2, https://github.com/leaanthony/go-webview2/blob/00f20a61c17da08a8ed60327e10a4d01f7008de1/pkg/edge/chromium.go#L76-L85

If it's invalid, it will be handled by webview2 and display a messagebox afterwards embed will fail. Do you want to add another check in wails?

**Comment by leaanthony** [comment]
Nah, I think that's ok. So long as users get informed. Anything but the default could lead to permission errors

**Comment by stffabi** [comment]
I think most often users want to change just the folder name in appdata to something more meaningful, instead of having the name of the binary in there.
That's the case for me 😀

**Comment by stffabi** [comment]
And thanks so much for merging

**Pull request merged**